core/src/main/java/com/radixdlt/consensus/sync/BFTSync.java

```diff
@@ -215,13 +215,13 @@ public EventProcessor<RoundQuorumReached> roundQuorumReachedEventProcessor() {
       final var highQC =
           switch (roundQuorumReached.votingResult()) {
             case FormedQC formedQc -> HighQC.from(
```

Comment:
Does it make sense for HighQC to have a `withHigherQC` / `withHigherTC` method on it which creates a new high qc?

Reply:
how about `withHighestTC` and `withHighestQC`, without any additional assumptions whether the qc/tc is indeed higher than the current?

Reply:
Sure, sounds good!

Perhaps withPossiblyHigherTC / withPossiblyHigherQC might be even more useful, but I'll let you decide 👍 :)